src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -98,11 +98,24 @@ public static void PortNumberInSPNTestForTCP()
             Assert.True(port > 0, "Named instance must have a valid port number.");
             builder.DataSource = $"{builder.DataSource},{port}";
 
-            PortNumberInSPNTest(builder.ConnectionString, port);
+            PortNumberInSPNTest(connectionString: builder.ConnectionString, expectedPortNumber: port);
+        }
+
+        [ConditionalFact(nameof(IsSPNPortNumberTestForNP))]
+        public static void SPNTestForNPMustReturnNamedInstance()
+        {
+            string connectionString = DataTestUtility.NPConnectionString;
```

Comment:
If you're not really making a connection to target database, this should be a Functional (Unit) test where you provide deterministic connection string properties and compare with expected results from the function called.

Reply:
The unit test depends on the config.json that is only available in Manual test. I'm not sure how to get the DataTest.Utility within the Functional Test.

Reply:
That's the point, it should not depend on Config.json if you're not actually connecting to anything. You can supply any dummy connection string in functional tests like we do in unit testing.

Reply:
I'm sorry if I am misunderstanding this unit test.  However, what I know is that the unit test needs to get the result of the driver's GetSqlServerSPN function from an actual physical named instance sql server. Hence, it needs the named instance data source from config.json.  I will find out if it is possible to hard code the actual physical named instance sql server data source in a dummy connection string then I will change it to functional test.

Reply:
@cheenamalhotra While the test doesn't connect to the database, it does use DNS to look up the server's FQDN in order to resolve the SPN for integrated security. It needs a real server for that.

Perhaps we should add code comments to make that clear.

Reply:
That makes sense, I didn't see the `GetSPNInfo()` before.. was mainly looking at testing instance name from the PR context. No worries, thanks for clarifying.